src/pages/academy/groundControl/subcomponents/GroundControlConfigureCell.tsx

```diff
@@ -0,0 +1,156 @@
+import {
+  Collapse,
+  Dialog,
+  DialogBody,
+  DialogFooter,
+  Divider,
+  Intent,
+  NumericInput,
+  Switch
+} from '@blueprintjs/core';
+import { IconNames, Team } from '@blueprintjs/icons';
+import React, { useCallback, useState } from 'react';
+
+import { AssessmentOverview } from '../../../../commons/assessment/AssessmentTypes';
+import ControlButton from '../../../../commons/ControlButton';
+
+type Props = {
+  handleConfigureAssessment: (
+    id: number,
+    hasVotingFeatures: boolean,
+    hasTokenCounter: boolean
+  ) => void;
+  data: AssessmentOverview;
+};
+
+const ConfigureCell: React.FC<Props> = ({ handleConfigureAssessment, data }) => {
+  const [isDialogOpen, setDialogState] = useState(false);
+  const [hasVotingFeatures, setHasVotingFeatures] = useState(!!data.hasVotingFeatures);
+  const [hasTokenCounter, setHasTokenCounter] = useState(!!data.hasTokenCounter);
+  const [isTeamAssessment, setIsTeamAssessment] = useState(false);
+
+  const handleOpenDialog = useCallback(() => setDialogState(true), []);
+  const handleCloseDialog = useCallback(() => setDialogState(false), []);
+
+  const handleConfigure = useCallback(() => {
+    const { id } = data;
+    handleConfigureAssessment(id, hasVotingFeatures, hasTokenCounter);
+    handleCloseDialog();
+  }, [data, handleCloseDialog, handleConfigureAssessment, hasTokenCounter, hasVotingFeatures]);
+
+  const toggleHasTokenCounter = useCallback(
+    () => setHasTokenCounter(!hasTokenCounter),
+    [hasTokenCounter]
+  );
+  const toggleVotingFeatures = useCallback(
+    () => setHasVotingFeatures(!hasVotingFeatures),
+    [hasVotingFeatures]
+  );
+  const toggleIsTeamAssessment = useCallback(
+    () => setIsTeamAssessment(!isTeamAssessment),
+    [isTeamAssessment]
+  );
```

Comment:
Nit: Here, we memoize, but we recreate every time `isXXXX` changes; this is not very ideal.

When toggling booleans, if we use the callback notation for `setXXX`, we can simply create the function once and not have to recreate it again:

```suggestion
  const toggleHasTokenCounter = useCallback(
    () => setHasTokenCounter(prev => !prev),
    []
  );
  const toggleVotingFeatures = useCallback(
    () => setHasVotingFeatures(prev => !prev),
    []
  );
  const toggleIsTeamAssessment = useCallback(
    () => setIsTeamAssessment(prev => !prev),
    []
  );
```